thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js

```diff
@@ -34,47 +34,79 @@ export default Ember.Component.extend({
   isLoading: false,
 
   /**
-   * List of metric urns, sorted by sortMode.
+   * List of metric urns, sorted by sortProperty and ordered by outputMode.
    */
   urns: Ember.computed(
     'entities',
     'metrics',
     'datasets',
     'changes',
     'sortProperty',
-    'sortMode',
+    'outputMode',
     function () {
-      const { entities, metrics, datasets, changes, scores, sortProperty, sortMode } =
-        this.getProperties('entities', 'metrics', 'datasets', 'changes', 'scores', 'sortProperty', 'sortMode');
+      const { entities, metrics, datasets, changes, scores, sortProperty, outputMode } =
+        this.getProperties('entities', 'metrics', 'datasets', 'changes', 'scores', 'sortProperty', 'outputMode');
 
       const metricUrns = filterPrefix(Object.keys(entities), ['thirdeye:metric:']);
-      let output = [];
 
-      switch(sortProperty) {
-        case ROOTCAUSE_METRICS_SORT_PROPERTY_METRIC:
-          output = metricUrns.map(urn => [metrics[urn], urn]).sort();
-          break;
+      // sort tuples
+      const tuples = metricUrns.map(urn => Object.assign({}, {
+        metric: metrics[urn],
+        dataset: datasets[urn],
+        change: makeSortable(changes[urn]),
+        score: makeSortable(scores[urn]),
+        urn
+      }));
 
-        case ROOTCAUSE_METRICS_SORT_PROPERTY_DATASET:
-          output = metricUrns.map(urn => [datasets[urn], urn]).sort();
-          break;
+      const sortedTuples = this._sort(tuples, sortProperty);
 
-        case ROOTCAUSE_METRICS_SORT_PROPERTY_CHANGE:
-          output = metricUrns.map(urn => [makeSortable(changes[urn]), urn]).sort((a, b) => a[0] - b[0]);
-          break;
+      const outputTuples = this._output(sortedTuples, outputMode);
 
-        case ROOTCAUSE_METRICS_SORT_PROPERTY_SCORE:
-          output = metricUrns.map(urn => [makeSortable(scores[urn]), urn]).sort((a, b) => a[0] - b[0]);
-          break;
-      }
+      return outputTuples.map(t => t.urn);
+    }
+  ),
 
-      if (sortMode === ROOTCAUSE_METRICS_SORT_MODE_DESC) {
-        output = output.reverse();
-      }
+  /**
+   * Sorts an array of tuples depending on sortProperty.
+   *
+   * @param {Array} tuples sort tuples
+   * @param {String} sortProperty sort property
+   * @returns {Array} sorted tuples
+   * @private
+   */
+  _sort(tuples, sortProperty) {
```

Comment:
nit: I would suggest naming it `sortBy` because you're providing an option for `sortProperty`